### ISSUE TITLE
Parity taking days to (fail to) sync on warp-barrier

### ISSUE THREAD

**Issue by cachvico** [paritytech/parity#8531]
I'm running:
>
> - **Which Parity version?**: `v1.10.2-beta-f4ae813-20180423/x86_64-linux-gnu/rustc1.25.0`
> - **Which operating system?**: `Ubuntu 16.06` (recently installed)
> - **How installed?**: `sudo bash <(curl https://get.parity.io -kL)`
> - **Are you fully synchronized?**: no
> - **Which network are you connected to?**: ethereum (default)
> - **Did you try to restart the node?**: yes

Parity taking days to (fail to) sync despite multiple restarts as follows:

1) Initial run started with canonical `parity` - cancelled after a couple of days, deleted `~/.local/share/io.parity.ethereum`

2) `parity --mode active --tracing off --pruning fast --db-compaction ssd --cache-size 1024` - again cancelled, deleted parity dir and then:

3) `parity --db-compaction ssd --cache-size 1024 --warp-barrier 5520000`

This last execution was left running for 48 hours and the last line of the log at that point:

`2018-05-02 15:52:13  Syncing #4179651 a017…7492     5 blk/s  434 tx/s  28 Mgas/s      0+  453 Qed  #4180107   25/25 peers   66 MiB chain 393 MiB db 43 MiB queue 13 MiB sync  RPC:  0 conn,  0 req/s, 44057 µs`

My system specs are as follows:

CPU: Intel i7 6700K
GPU: GeForce GTX 1060 6GB
RAM: 16GB DDR4 SDRAM
HD: 128GB M.2 SSD
Internet: 950 Mb/s down, 120 Mb/s up, fiber
Clock is synced. 8GB swapfile is 152M used.

StackExchange discussion here, https://ethereum.stackexchange.com/questions/46950/parity-sync-time-april-2018-many-days-normal

Raising a bug as requested by Tbaut. I would be happy to help with diagnosis and testing.

Thank you,

Darren

**Comment by cachvico** [comment]
Having deleted the data dir and restarted, I'm going to copy here the first page of output. Perhaps there will be something obviously wrong here (to those in the know):

```
darren@dubuntu:~$ parity --warp-barrier 5520000
2018-05-02 16:15:17  Starting Parity/v1.10.2-beta-f4ae813-20180423/x86_64-linux-gnu/rustc1.25.0
2018-05-02 16:15:17  Keys path /home/darren/.local/share/io.parity.ethereum/keys/Foundation
2018-05-02 16:15:17  DB path /home/darren/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-05-02 16:15:17  Path to dapps /home/darren/.local/share/io.parity.ethereum/dapps
2018-05-02 16:15:17  State DB configuration: fast
2018-05-02 16:15:17  Operating mode: active
2018-05-02 16:15:17  Configured for Foundation using Ethash engine
2018-05-02 16:15:17  Updated conversion rate to Ξ1 = US$674.47 (7060217.5 wei/gas)
2018-05-02 16:15:22  Public node URL: enode://1fc338abaac538bbac456806d14515e5478712425f6586b0c384c7d13a83a127d16bf5bfac846f1fb76e69686fb5946eb5a5214d8bd0cb42b5a89350bfac421b@192.168.0.125:30303
2018-05-02 16:15:37  Syncing snapshot 0/1315        #0    6/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:15:42  Syncing snapshot 0/1315        #0    6/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:15:52  Syncing snapshot 6/1315        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:15:57  Syncing snapshot 12/1315        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:16:03  Syncing snapshot 17/1315        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:16:12  Syncing snapshot 24/1315        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:16:23  Syncing snapshot 29/1315        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:16:57  Syncing snapshot 31/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:11  Syncing snapshot 38/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:21  Syncing snapshot 44/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:27  Syncing snapshot 46/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:32  Syncing snapshot 49/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:37  Syncing snapshot 50/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:47  Syncing snapshot 51/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:17:57  Syncing snapshot 51/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:02  Syncing snapshot 51/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:12  Syncing snapshot 51/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:22  Syncing snapshot 51/1315        #0    8/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:32  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:42  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:18:52  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:02  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:07  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:17  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:27  Syncing snapshot 51/1315        #0   11/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:37  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:47  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:19:57  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:07  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:17  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:27  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:37  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:47  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:20:57  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:07  Syncing snapshot 51/1315        #0    9/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:17  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:27  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:37  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:47  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:21:57  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:22:02  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-02 16:22:12  Syncing snapshot 51/1315        #0   10/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
...
2018-05-02 16:28:12  Syncing snapshot 51/1315        #0   16/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
...
```

**Comment by Tbaut** [comment]
cc: @dvdplm, @ngotchac and  @tomusdrw

**Comment by JSterling8** [comment]
Same here. Seems to start with warp sync, then switches to full sync.

> - Which Parity version?: v1.10.2-beta-f4ae813-20180423/x86_64-linux-gnu/rustc1.25.0
> - Which operating system?: Ubuntu 16.06
> - How installed?: Downloaded .deb from GitHub -> Ubuntu Software Manager installed
> - Are you fully synchronized?: no
> - Which network are you connected to?: ethereum (default)
> - Did you try to restart the node?: yes

My system specs are as follows:

CPU: Intel® Core™ i7-7500U CPU @ 2.70GHz × 4 
GPU: N/A
RAM: 16GB
HD: 512GB
Internet: 50 Mb/s down, 20 Mb/s up, fiber

Output:

```
➜  ~ git:(master) ✗ parity --db-compaction ssd --cache-size 1024 --warp-barrier 5520000 ui
2018-05-06 16:52:35  Parity browser interface is deprecated. It's going to be removed in the next version, use standalone Parity UI instead.
2018-05-06 16:52:35  Standalone Parity UI: https://github.com/Parity-JS/shell/releases
2018-05-06 16:52:35  Starting Parity/v1.10.2-beta-f4ae813-20180423/x86_64-linux-gnu/rustc1.25.0
2018-05-06 16:52:35  Keys path /home/jsterling8/.local/share/io.parity.ethereum/keys/Foundation
2018-05-06 16:52:35  DB path /home/jsterling8/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-05-06 16:52:35  Path to dapps /home/jsterling8/.local/share/io.parity.ethereum/dapps
2018-05-06 16:52:35  State DB configuration: fast
2018-05-06 16:52:35  Operating mode: active
2018-05-06 16:52:35  Configured for Foundation using Ethash engine

Open: http://127.0.0.1:8180/#/auth?token=<REDACTED>
to authorize your browser.
Or use the generated token:
<REDACTED>
2018-05-06 16:52:36  Public node URL: enode://3108a412f3b3f8c2001802ca22aa8c81aa8a3f22536d5839379c44354a06d23fd82af268ca2067c1bce5e553890c95da514cc03e6063e5c5c2d10412db5847e8@192.168.1.155:30303
Created new window in existing browser session.
[657:670:0506/165236.412743:ERROR:browser_gpu_channel_host_factory.cc(119)] Failed to launch GPU process.
2018-05-06 16:52:37  Updated conversion rate to Ξ1 = US$774.46 (6148677 wei/gas)
2018-05-06 16:53:01  Syncing snapshot 0/1337        #0    2/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  1 conn,  1 req/s,  51 µs
2018-05-06 16:53:11  Syncing snapshot 2/1337        #0    3/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  1 conn,  1 req/s,  51 µs
...
2018-05-06 16:55:41  Syncing snapshot 21/1337        #0    7/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  1 conn,  2 req/s, 198 µs
2018-05-06 16:55:51  Syncing       #0 d4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s    123+    0 Qed      #127    4/25 peers   8 KiB chain 3 MiB db 234 KiB queue 73 KiB sync  RPC:  1 conn,  1 req/s, 229 µs
2018-05-06 16:55:56  Syncing    #1524 3d94…7253   304 blk/s    0 tx/s   0 Mgas/s      2+    0 Qed     #1527    6/25 peers   2 MiB chain 11 MiB db 4 KiB queue 419 KiB sync  RPC:  1 conn,  4 req/s, 198 µs
2018-05-06 16:56:06  Syncing    #1781 2f92…6a9a    25 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed     #1781    7/25 peers   2 MiB chain 11 MiB db 0 bytes queue 5 MiB sync  RPC:  1 conn,  3 req/s,  80 µs
2018-05-06 16:56:16  Syncing    #1781 2f92…6a9a     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed     #1781    7/25 peers   2 MiB chain 11 MiB db 0 bytes queue 9 MiB sync  RPC:  1 conn,  1 req/s,  80 µs
```

**Comment by JSterling8** [comment]
CTRL+C'ing then rerunning with the same command appears to be working now. At snapshot 365 of 1337 at present. Will update if it completes or switches back to full syncing again.

**Comment by JSterling8** [comment]
Just got a full sync. Total time about 2.5 hours. @cachvico - Instead of deleting `~/.local/share/io.parity.ethereum` and restarting, simply CTRL+C then run again with the same params. I used: `parity --db-compaction ssd --cache-size 1024 --warp-barrier 5520000`

**Comment by cachvico** [comment]
I have restarted first with `parity` (accidentally) then `parity --warp-barrier 5520000`. I've updated my log (in the second post in this thread) with the complete output appended.

**Comment by JSterling8** [comment]
@cachvico - Does it get further for you now after the restart? Mine didn't complete the first time, but completed on the first restart.

**Comment by cachvico** [comment]
@JSterling8 not really - still only up to #4638589

I've updated my log at https://github.com/paritytech/parity/issues/8531#issuecomment-386110738

**Comment by cachvico** [comment]
And we're off! Fingers crossed

**Comment by cachvico** [comment]
I deleted `~/.local/share/io.parity.ethereum`, built from source using `~/.cargo/bin/cargo build --release` on branch `ng-warp-resume`, and invoked `./target/release/parity  --db-compaction ssd --cache-size 1024 --warp-barrier 5520000`.

Observe that I still seemed to need to hit Ctrl-C and restart, in order for sync from snapshots to work:

```
$ ./target/release/parity  --db-compaction ssd --cache-size 1024 --warp-barrier 5520000
2018-05-08 12:01:20  Starting Parity/v1.12.0-unstable-ce2bce412-20180507/x86_64-linux-gnu/rustc1.25.0
2018-05-08 12:01:20  Keys path /home/darren/.local/share/io.parity.ethereum/keys/Foundation
2018-05-08 12:01:20  DB path /home/darren/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-05-08 12:01:20  Path to dapps /home/darren/.local/share/io.parity.ethereum/dapps
2018-05-08 12:01:20  State DB configuration: fast
2018-05-08 12:01:20  Operating mode: active
2018-05-08 12:01:20  Configured for Foundation using Ethash engine
2018-05-08 12:01:21  Sending warning alert CloseNotify
2018-05-08 12:01:21  Updated conversion rate to Ξ1 = US$733.34 (6493447.5 wei/gas)
2018-05-08 12:01:26  Public node URL: enode://d270659d99405b9dffc3d397d8e97a4dc84759f2b1beea0af61a809711d21bb059d3f893bb0017c31c1a42b8d01d49625548f71d3b9e3794b7aebf4792dde7d4@192.168.0.125:30303
2018-05-08 12:01:31  Syncing       #0 0xd4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed        #0    4/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:01:41  Syncing       #0 0xd4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed        #0    3/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:01:51  Syncing       #0 0xd4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed        #0    3/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
...
... (skip 30 mins)
...
2018-05-08 12:35:21  Syncing       #0 0xd4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed        #0   25/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:35:26  Syncing       #0 0xd4e5…8fa3     0 blk/s    0 tx/s   0 Mgas/s      0+    0 Qed        #0   25/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
^C2018-05-08 12:35:31  Finishing work, please wait...
$ 
$ ./target/release/parity  --db-compaction ssd --cache-size 1024 --warp-barrier 5520000
2018-05-08 12:35:33  Starting Parity/v1.12.0-unstable-ce2bce412-20180507/x86_64-linux-gnu/rustc1.25.0
2018-05-08 12:35:33  Keys path /home/darren/.local/share/io.parity.ethereum/keys/Foundation
2018-05-08 12:35:33  DB path /home/darren/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-05-08 12:35:33  Path to dapps /home/darren/.local/share/io.parity.ethereum/dapps
2018-05-08 12:35:33  State DB configuration: fast
2018-05-08 12:35:33  Operating mode: active
2018-05-08 12:35:33  Configured for Foundation using Ethash engine
2018-05-08 12:35:34  Sending warning alert CloseNotify
2018-05-08 12:35:34  Updated conversion rate to Ξ1 = US$733.80 (6489376.5 wei/gas)
2018-05-08 12:35:39  Public node URL: enode://d270659d99405b9dffc3d397d8e97a4dc84759f2b1beea0af61a809711d21bb059d3f893bb0017c31c1a42b8d01d49625548f71d3b9e3794b7aebf4792dde7d4@192.168.0.125:30303
2018-05-08 12:35:49  Syncing snapshot 0/1344        #0    5/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:35:59  Syncing snapshot 0/1344        #0   12/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:04  Syncing snapshot 0/1344        #0   12/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:14  Syncing snapshot 0/1344        #0   16/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:19  Syncing snapshot 1/1344        #0   20/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:29  Syncing snapshot 3/1344        #0   22/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:34  Syncing snapshot 4/1344        #0   20/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2018-05-08 12:36:44  Syncing snapshot 4/1344        #0   19/25 peers   8 KiB chain 3 MiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
...
```

**Comment by cachvico** [comment]
In fact it seems to choose randomly between snapshot sync and regular sync when restarting after ctrl-c

**Comment by ngotchac** [comment]
@cachvico Actually the logs saying "Syncing #0" is quite misleading because it could be that the node is just waiting for valid peers to start the snapshot. So the issue here is that it can't find one.
I could see two reasons : 
1. The peer is blocking the connection or not answering the requests
2. There are already too many other peers connected (and that's something we could/should fix)

Just to be sure, if you could provide `-l sync,warp,snapshot,network=trace` logs it would be great!

**Comment by ngotchac** [comment]
(The issue on the mainnet is that currently there seem to be only one peer serving snapshots)

**Comment by cachvico** [comment]
Obviously the `$ ./target/release/parity  --db-compaction ssd --cache-size 1024 --warp-barrier 5520000 -l sync,warp,snapshot,network=trace` command generates gobs of logs - how many minutes-worth would you like me to send?

**Comment by ngotchac** [comment]
Yeah sure, just the first minute should be fine.

**Comment by cachvico** [comment]
Here we go:

[output.txt](https://github.com/paritytech/parity/files/1984952/output.txt)

**Comment by ngotchac** [comment]
So reading from the logs, it seems that the only peer that could provide a snapshot (`0x1d1f7...`) closes the TCP connection as soon as a Handshake is initialized (tried 3 times in your run).
I'm not sure why that is, it could be that it already has too many open connections (cc @arkpar ?).
A potential solution would be to delete your `~/.local/share/io.parity.ethereum/chains/ethereum/network` folder, and also try running with `--no-discovery`

**Comment by cachvico** [comment]
Can I just check that there aren't any LAN port-forward requirements?

**Comment by cachvico** [comment]
I deleted my network folder and restarted with `--no-discover`.

I've been trying to sync with Parity for 10 days now.. should I restart *without* `--warp-barrier` if it has a greater chance of success?

**Comment by cachvico** [comment]
Restarted without warp, in branch `ng-warp-resume`, and after 8 hours, finally synced! 

\o/

**Comment by ngotchac** [comment]
Daily update : we actually found 2 bugs you might have run into:
1. The remote node would disconnect while your node was fetching the snapshot manifest, so your node would be stuck indefinitely. This is now fixed in `ng-warp-resume` #8544 
2. The remote node would disconnect every 3.5 seconds. The fix is here #8555 or you can fix it manually by running `--no-serve-light`

**Comment by 5chdn** [comment]
The 2nd one should not be on 1.10 branch.

**Comment by folsen** [comment]
Closing this issue as the latest PRs should have addressed it. Reopen if it persists.

**Comment by cachvico** [comment]
I've been using Parity since my successful sync two weeks ago, but didn't have too much luck on a clean restart today.

Can I confirm the fixes are merged into `master`? I deleted `~/.local/share/io.parity.ethereum`, switched to master, pulled, built and restarted with `./target/release/parity --db-compaction ssd --cache-size 1024 --warp-barrier 5650000`, but it's not looking too promising - I've attached a log file showing various restarts over a couple of hours.

Thanks a lot

[output.txt](https://github.com/paritytech/parity/files/2029299/output.txt)

**Comment by folsen** [comment]
@cachvico The first two times you're shutting your node down while it's trying to download a snaphot? And it's actually successfully picking up where it left off. The last time I think you're being hit by a bug that's only on master that's a deadlock issue, it's being fixed right now (in general building and running from master is not always the most reliable).

**Comment by cachvico** [comment]
Thanks @folsen - which is the branch to build from to pick up all the fixes referenced in this issue?

**Comment by folsen** [comment]
@cachvico I don't know, it depends on what you're trying to do, are you trying to develop something and work on a fix for something? Otherwise I'd just recommend to download the latest release.

**Comment by cachvico** [comment]
I'm simply trying to run the Parity client (e.g. for blockchain querying via web3). If all the above fixes are rolled into the latest release, I'll give that one a go.

**Comment by cachvico** [comment]
I'm simply trying to run the Parity client (e.g. for blockchain querying via web3).

Sorry can you confirm which release version to use? My build is on 1.12.0-unstable, but the latest release on the website is 1.10.4. Given all the problems I had earlier (hence this Issue thread), I don't want to try again with the wrong version..

**Comment by folsen** [comment]
Try 1.11.1 from here https://github.com/paritytech/parity/releases

**Comment by cachvico** [comment]
Built 1.11.1 from source and it's running, however not sure it's working. Output attached - first run from clean build after 2 hours.

Couple of more minor observations:

- Is there a newcomer guide to all the columns in the standard output? Half of the chit-chat in this bug report has (unfortunately) been due to me not having a clue what's going on - I see a bunch of zeros but have basically no idea what's happening (in particular whether the fast sync / warp barrier etc. stuff is working).
- Is there an "uninstall" guide anywhere? My first install was done with the one-line rustup command (for Ubuntu). Since then, I have no idea what's in my system and needs to be deleted to be able to uninstall / clean install a new version. (That's why I built 1.11.1 from source). Is `/usr/bin/parity` the only file installed, plus `~/.local/share/io.parity.ethereum`? (Even if it's a quick rustup command, that would be useful to provide for those who don't know Rust/rustup)
- Most importantly - is the attached output showing normal state?

Many thanks for all the follow-ups.

[output.txt](https://github.com/paritytech/parity/files/2032811/output.txt)

**Comment by Tbaut** [comment]
@cachvico thanks for your feedback. A newcomer guide isn't available right now but a work to make [our FAQ](https://github.com/paritytech/wiki/issues/93) more beginner friendly has already started.

For such cases as yours, please join our [Gitter channel](https://gitter.im/paritytech/parity) to get to understand how things work under the hood, not only from the parity team but also from the community and more importantly help you resolve your problems.

**Comment by cachvico** [comment]
Sorry what does "cases like mine" mean?

Please note that since opening this Github Issue (at your request @Tbaut on StackOverflow) *I still don't have a functioning Parity client* despite following instructions and due diligence with clean builds, etc.

Ever so slightly frustrating to be given the brush-off here.

**Comment by folsen** [comment]
@cachvico A case like yours would be where someone despite (your) best efforts can't get parity running properly, we've had this back-and-forth in this issue for over a month. Most people don't have this experience, and having one message back and forth per day for another month is not the most effective way to deal with it. If you jump into chat with us, we can probably get it resolved faster.

**Comment by cachvico** [comment]
To be clear, I was running the fix set up for me by yourselves, for 2 weeks of that month. You then (reasonably) closed the bug but I was stranded on a dev branch so now all I need to do is build from the fixed release.

Except the fixed release given is not working, so back to square 1 (in fact a step backward because now I don't have a functioning node at all, which I did before).

I can now be reached on the Gitter channel and will post a message in due course..

**Comment by folsen** [comment]
@cachvico Yeah I feel you, you've been hit by a cacophony of different issues, all of them should have been fixed, but I think you may just need some patience until we cut a 1.11.2 release to get all of them. Building from master you hit another issue and we don't recommend running a node that is built from master. You'd probably have more luck going back to 1.9.x or 1.10.x or some stable release than trying to rely on beta/alpha builds. 1.11.1 _should_ work for you, but for some reason your sync got stuck, there can be a _ton_ of reasons for this, thus it's easier to debug on chat. Hopefully it'll be resolved shortly.

**Comment by cachvico** [comment]
Tbaut verified my above results in chat, confirmed fast sync from clean start is currently unavailable, and advised me to keep retrying.

I'm trying `--no-warp`, which sounds like it will beat the hare at this rate.

**Comment by Tbaut** [comment]
@cachvico It's related to https://github.com/paritytech/parity/pull/8710 to be patched in 1.11.2

**Comment by cachvico** [comment]
OK. I'm back on geth in the meantime, completed a full sync in about two days.
